src/SprykerEco/Zed/Adyen/Business/Hook/AdyenPostSaveHook.php

```diff
@@ -77,7 +87,7 @@ public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $
 
         if (
             !$responseTransfer->getIsSuccess()
-            || $responseTransfer->getMakePaymentResponse()->getResultCode() === static::ADYEN_OMS_STATUS_REFUSED
+            || in_array($responseTransfer->getMakePaymentResponse()->getResultCode(), static::ADYEN_OMS_STATUS_REFUSAL_REASONS) == true
```

Comment:
please set 3rd arg to true for in_array() calls.

Reply:
and instead, the === true part at the end is not necessary and should be removed.

Reply:
changed

Reply:
The 3rd arg has not been adjusted yet.

Reply:
sorry, added strict for type check